chatbot-core/rag/embedding/bm25_indexer.py

```diff
@@ -0,0 +1,79 @@
+"""
+Module for the Sparse Retriever Class.
+"""
+
+from retriv import SparseRetriever
+from utils import LoggerFactory
+
+# pylint: disable=too-few-public-methods
+class BM25Indexer:
+    """
+    Class that represents the indexer for the bm25 Sparse Retriever.
+    """
+    def __init__(self, index_configs, logger):
+        """
+        Initialize with a list of index configurations.
+        Each config should be a dict with keys: 'index_name' and 'file_path'.
+        """
+        self.logger = logger
+        self.index_configs = index_configs
+        self.retrievers = {}
+        self._create_indexes()
+
+    def _create_indexes(self):
```

Comment:
The `_create_indexes` method is doing both indexing and retriever object assignment, which couples concerns. Consider separating indexing into a distinct method or utility function if batch indexing becomes more complex. also it runs immediately during object instantiation. Consider deferring it via a method like .build() so the class can be instantiated independently from the heavy task of indexing, which also helps with testability.